custom_components/elmo_iess_alarm/helpers.py

```diff
@@ -67,3 +68,12 @@ async def validate_credentials(hass: core.HomeAssistant, config: dict):
     client = ElmoClient(config.get(CONF_SYSTEM_URL), domain=config.get(CONF_DOMAIN))
     await hass.async_add_executor_job(client.auth, config.get(CONF_USERNAME), config.get(CONF_PASSWORD))
     return True
+
+def generate_entity_name(entry: ConfigEntry) -> str:
```

Comment:
This is great!

I would recommend to add an optional kwarg so that if's passed, it adds the `name` argument. The signature becomes:
```python
def generate_entity_name(entry: ConfigEntry, name: str = None) -> str:
    pass
```

Then in the code you use that name if present or empty otherwise.

In terms of docstring, you can add the following (feel free to remove the `#` comment as it is now included in the docstring):
```python
"""Generate a name for the entity based on system configuration or username.

    Args:
        entry (ConfigEntry): The configuration entry from Home Assistant containing system configuration or username.

    Returns:
        str: The generated entity name, which is a combination of the domain and either the configured system name or the username.

    Example:
        >>> entry.data = {"system_name": "Seaside Home"}
        >>> generate_entity_name(entry, "window")
        "elmo_iess_alarm_seaside_home_window"
    """
```

I think you wanted it to show up as `elmo_iess_alarm_seaside_home_window` right?